src/components/ArticleDetailPage.jsx

```diff
@@ -0,0 +1,43 @@
+import React, {useState, useEffect} from 'react'
+import { useParams } from 'react-router-dom'
+import axios from 'axios'
+
+const ArticleDetailPage = () => {
+    const {article_id} = useParams()
+    const [article, setArticle] = useState(null)
+    const [loading, setLoading] = useState(true)
+    const [error, setError] = useState(null)
+
+    useEffect(() => {
+        axios.get(`https://cwazycodes-nc-news.onrender.com/api/articles/${article_id}`)
+        .then((response) => {
+            setArticle(response.data.article)
+            setLoading(false)
+        })
+        .catch((err) => {
+            console.error(err)
+            setError(err.message)
+            setLoading(false)
+        })
+    }, [article_id])
```

Comment:
lets think about if we need to pass anything in this dependency array 🙂. Think about the how we get to this page and when we change the endpoint to view another article what happens then.